pkg/sql/catalog/resolver/resolver.go

```diff
@@ -274,6 +274,56 @@ func ResolveSchemaNameByID(
 	return "", errors.Newf("unable to resolve schema id %d for db %d", schemaID, dbID)
 }
 
+// ResolveSchemaDescByID resolves a schema based on it's ID.
+func ResolveSchemaDescByID(
+	ctx context.Context, txn *kv.Txn, codec keys.SQLCodec, schemaID descpb.ID,
+) (sqlbase.ResolvedSchema, error) {
+	if schemaID == keys.PublicSchemaID {
+		return sqlbase.ResolvedSchema{
+			Kind: sqlbase.SchemaPublic,
+			ID:   schemaID,
+		}, nil
+	}
+
+	// We have already considered if the schemaID is PublicSchemaID,
+	// if the id appears in staticSchemaIDMap, it must map to a virtual schema.
+	if _, ok := staticSchemaIDMap[schemaID]; ok {
+		return sqlbase.ResolvedSchema{
+			Kind: sqlbase.SchemaVirtual,
+			ID:   schemaID,
+		}, nil
+	}
+
+	// We have already considered public schemas and virtual schemas.
+	// The schema must be user-defined or temporary.
+	desc, err := catalogkv.GetDescriptorByID(ctx, txn, codec, schemaID, catalogkv.Immutable,
+		catalogkv.SchemaDescriptorKind, false)
+	if err != nil {
+		return sqlbase.ResolvedSchema{}, err
+	}
+
+	// TODO(richardjcai): We should store temp schema IDs when we create it in
+	// the session data.
+	// For now assume a nil descriptor corresponds to a temp table.
+	if desc == nil {
```

Comment:
I'd rather just not handle the temporary schema case here than assume nil == temp. Leaving the TODO is fine however, but assign it to SQLSchema